expose `Conf` and `P2P`

Declaring a conf variable as `lightningd::conf::Conf::default();` fails with error:
```
   |
19 | mod conf;
   | ^^^^^^^^

error[E0603]: module `conf` is private
   --> src/main.rs:101:28
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -8,8 +8,8 @@ use std::{
 
 use bitcoind::BitcoinD;
 use clightningrpc::LightningRPC;
-use conf::Conf;
-use error::Error;
+pub use conf::Conf;
+pub use error::Error;
 use log::debug;
 use tempfile::TempDir;
 

